detection_rules/rule_formatter.py

```diff
@@ -161,7 +161,7 @@ def dump_list(self, v):
                     dump.append(' ' * 4 + self.dump_value(item))
             return '[\n{},\n]'.format(',\n'.join(dump))
 
-        if v and all(isinstance(i, dict) for i in v):
+        if all(isinstance(i, dict) for i in v):
```

Comment:
I think we can make it clearer! Reading the code, it's not obvious that this check will return `True` for an empty list / iterator, but it will:

```
In [1]: all([])
Out[1]: True

In [2]: all(x for x in [])
Out[2]: True
```

If that's desirable, we can make it explicit by doing

```
if len(v) == 0 or all(isinstance(i, dict) for i in v):
```